custom_components/babybuddy/config_flow.py

```diff
@@ -0,0 +1,180 @@
+    ) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        self._reauth_unique_id = self.context["unique_id"]
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Dialog that informs the user that reauth is required."""
+        errors = {}
+        existing_entry = await self.async_set_unique_id(self._reauth_unique_id)
+        if user_input is not None and existing_entry is not None:
+            user_input[CONF_HOST] = existing_entry.data[CONF_HOST]
+            user_input[CONF_PORT] = existing_entry.data[CONF_PORT]
+            try:
+                client: BabyBuddyClient = BabyBuddyClient(
+                    user_input[CONF_HOST],
+                    user_input[CONF_PORT],
+                    user_input[CONF_API_KEY],
+                    self.hass.helpers.aiohttp_client.async_get_clientsession(),
+                )
+                await client.async_connect()
+            except ConnectError:
+                errors["base"] = "cannot_connect"
+
+            if not errors:
+                self.hass.config_entries.async_update_entry(
+                    existing_entry,
+                    data={**existing_entry.data, **user_input},
+                )
+                await self.hass.config_entries.async_reload(existing_entry.entry_id)
+                return self.async_abort(reason="reauth_successful")
+
+        return self.async_show_form(
+            step_id="reauth_confirm",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_API_KEY): str,
+                }
+            ),
+            errors=errors,
+        )
+
+
+class BabyBuddyOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle Babybuddy options."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage Babybuddy options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        options = {
+            vol.Optional(
+                CONF_TEMPERATURE_UNIT,
+                default=self.config_entry.options.get(
+                    CONF_TEMPERATURE_UNIT, TEMP_CELSIUS
+                ),
+            ): vol.In([TEMP_CELSIUS, TEMP_FAHRENHEIT]),
+            vol.Optional(
+                CONF_WEIGHT_UNIT,
+                default=self.config_entry.options.get(CONF_WEIGHT_UNIT, MASS_KILOGRAMS),
+            ): vol.In([MASS_KILOGRAMS, MASS_POUNDS]),
+            vol.Optional(
+                CONF_FEEDING_UNIT,
+                default=self.config_entry.options.get(
+                    CONF_FEEDING_UNIT, VOLUME_MILLILITERS
+                ),
```

Comment:
I'm not sure it makes sense to designate a unit for either all children or all times across a child's timeline. I think this should be removed since Baby Buddy itself doesn't designate units.

Reply:
Yes Babybuddy doesn't use measurement units but in HA using measurement units allow for a better display of sensor data. This is not reflected in Baby buddy and the user can choose the unit that he prefers and change it at anytime.

Reply:
I think I'd like the default to be no unit_of_measurement to match BB, but then allow for CONFIG_ENTRY to determine.

Would ask for MASS_OUNCES to be added as option.